lib/PreviewProvider.js

```diff
@@ -173,40 +181,74 @@ PreviewProvider.prototype = {
       return links;
     }
 
-    return links.map(link => {
+    let now = Date.now();
+
+    let results = links.map(link => {
       if (link && link.cacheKey && ss.storage.embedlyData[link.cacheKey]) {
+        ss.storage.embedlyData[link.cacheKey].accessTime = now;
         return Object.assign({}, ss.storage.embedlyData[link.cacheKey], link);
       } else {
         return null;
       }
     })
     .filter(link => link);
+
+    // gives the opportunity to have at least one run with an old
+    // preview cache. This is for the case where one hasn't opened
+    // the browser since `cacheTTL`
+    this.cleanUpCacheMaybe();
```

Comment:
Yeah, I *hope* it won't take too much time.

That said, the periodic run has a chance to never run: if a user never keeps their browser open for longer than 24 hours, or whatever the timeout was.

One way to fix it would've been to run a cleanup job at startup, but that is also not desirable because a user may have not opened their browser for a long time.

We *could* solve the problem with timeouts, which would be better, but I thought we'd give this a try first.